From: Mikael Magnusson <mikma@users.sourceforge.net>
Subject: [PATCH] t/fix/naptr_replacement

Fix enum lookup on Erlang/OTP R13.
All ENUM NAPTR records decoded by R13 were ignored since it adds a
terminating "." which YXA doesn't expect.

Signed-off-by: Mikael Magnusson <mikma@users.sourceforge.net>

---
 src/dnsutil.erl |   24 +++++++++++++++++++++++-
 1 files changed, 23 insertions(+), 1 deletions(-)

diff --git a/src/dnsutil.erl b/src/dnsutil.erl
index 0f88a2b..3c526c2 100644
--- a/src/dnsutil.erl
+++ b/src/dnsutil.erl
@@ -816,9 +816,31 @@ naptr_from_tuple({Order, Preference, Flags, Services, Regexp, Replacement}) ->
 		 flags=Flags,
 		 services=string:to_upper(Services),
 		 regexp=Regexp,
-		 replacement=Replacement
+		 replacement=remove_last_period(Replacement)
 		}.
 
+
+%%--------------------------------------------------------------------
+%% @spec    (Replacement) ->
+%%            Replacement
+%%
+%%            Replacement = string()
+%%
+%% @doc     Removes the terminating period (".") of the replacement
+%%          string.
+%% @end
+%%--------------------------------------------------------------------
+remove_last_period(Replacement) ->
+    remove_last_period(Replacement, []).
+
+remove_last_period([], Acc) ->
+    Acc;
+remove_last_period([$.], Acc) ->
+    Acc;
+remove_last_period([E | Replacement], Acc) ->
+    remove_last_period(Replacement, [E | Acc]).
+    
+
 %%--------------------------------------------------------------------
 %% @spec    (Len, In, []) ->
 %%            Replacement
-- 
tg: (5aeee79..) t/fix/naptr_replacement (depends on: upstream)
